### PR TITLE
Interface with dust_extinction package + generalized RvFA Law

### PR DESCRIPTION
**Pull request by lcjohnso**
PR reflects work on Issue https://github.com/BEAST-Fitting/beast/issues/259.

Two items accomplished here:
1) Added generalized RvFA extinction law class (`extinction.Generalized_RvFALaw`) that allows user to pass assignments for A and B curves via class call.
2) Added wrapper / interface class (`extinction.Generalized_DustExt`) that allows use of classes defined in stand-alone [dust_extinction](https://dust-extinction.readthedocs.io) package.

Tests and documentation were added regarding new extinction classes.

Open Questions:

- Defining dependencies is a mystery to me.  I added dust_extinction in three different locations, but unsure which are necessary (if any).
- Here, work was meant to extend current capabilities and does not eliminate previous classes.  I was hesitant to swap in new dust_extinction classes in place of existing BEAST-specific classes until users have had time to experiment with new capabilities.

### CODE REVIEW

**Comment by karllark** [comment]
This pull request **introduces 2 alerts** when merging 27e15633a26e3fe40b4e213e1c51d00e9ff73bcf into a02e5fb4ae79fc2b6bd2ccfb13422a8f3f4cd60c - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-66971ed81c44d52df0cd48cae5269074d06db8dc)

**new alerts:**

* 2 for Missing call to \_\_init\_\_ during object initialization

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by karllark** [comment]
This pull request **introduces 2 alerts** when merging 2d560b7793d52d930052fd6b3bf5ad8b2cc00058 into a02e5fb4ae79fc2b6bd2ccfb13422a8f3f4cd60c - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-99b199b914b1076622ce2d2b4cdff3e96b421c1a)

**new alerts:**

* 2 for Missing call to \_\_init\_\_ during object initialization

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by karllark** [comment]
Looking good with a quick pass review.  Yeah! 
Are you planning on replacing the existing code?  Could be done and then the automated testing will confirm it is the same.  Running 'python setup.py tests --remote-data' will test this on your changes locally if you aren't already doing this.  :-)

**Comment by lcjohnso** [comment]
There will be some differences: it appears the F99 curve is different at few parts in 1.e-03 (in relative scaling) -- I varied tolerance on my limited tests to identify the scale of this offset.  Not sure whether that's significant in your mind or not, or how best to investigate / compare any offsets identified. Thoughts?

My primary goal here was to augment current capabilities, so I feel that is complete.  Would you prefer to rip out BEAST-specific code now and switch fully to dust_extinction?  Is that package stable enough to link as a core dependence?  This would be a real turning-point move, locking both packages in for the long-haul regarding syntax, packaging, etc.

**Comment by karllark** [comment]
I will investigate this next week.  Currently taking a few days off.  :-)

**Comment by karllark** [comment]
@lcjohnso : I've investigated the differences in the BEAST and dust_extinction implementations of the F99 R(V) dependent model.  The differences are not unexpected given the dust_extinction implementation strictly follows the F99 paper and the BEAST implementation follows the F99 IDL code.  The F99 IDL code was updated for unpublished/undocumented (to my knowledge) updates that Ed Fitzpatrick did sometime after the F99 paper.  For the dust_extinction package, I put a priority on traceability - so we strictly follow the published papers.  The differences between the IDL code and the F99 paper was not something I was fully aware of until working on the dust_extinction package and I was working hard to document everything carefully.

So, I think we are good.  In the end, we will probably should switch to only using models from dust_extinction, but this is a minor point right now.

**Comment by karllark** [comment]
The dependencies look good.

File: beast/physicsmodel/dust/extinction.py [hunk 1]
```diff
@@ -567,3 +571,87 @@ def get_Rv(self, Rv_A, f_A=0.5):
 
         Rv_B = self.BLaw.Rv
         return 1. / (f_A / Rv_A + (1 - f_A) / Rv_B)
+
+
+class Generalized_RvFALaw(Gordon16_RvFALaw):
```
Comment (karllark) [inline beast/physicsmodel/dust/extinction.py]: I think this should inherit ExtinctionLaw, not Gordon16_RvFALaw.  Unless you are using something from the Gordon16_RvFALaw.

**Comment by karllark** [comment]
@lcjohnso : if you can address my one comment, then I think this PR is ready for merging.

File: beast/physicsmodel/dust/extinction.py (hunk 1 above)
Comment (lcjohnso) [reply beast/physicsmodel/dust/extinction.py]: I'm inheriting Gordon16_RvFALaw in order to not duplicate the get_Rv and get_Rv_A functions.

**Comment by lcjohnso** [comment]
Re: Lea's comment about examples and implementation -- as this PR is pitched to add capabilities, but not fully replace existing dust extinction models / code, I will hold off on replacing extinction curve usage in datamodel.py until we are ready for full transition.

**Comment by karllark** [comment]
This pull request **introduces 2 alerts** when merging 5334756c35af17fbb08a69603d900a7fc5f2d32b into 7b06c434ecaedc58549eefa6588e16152efd3a7d - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-4c2ab215fcbb2f1a120f6f4a3969a11e4c7d3343)

**new alerts:**

* 2 for Missing call to \_\_init\_\_ during object initialization

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**